python_modules/dagster/dagster/_core/storage/partition_status_cache.py

```diff
@@ -0,0 +1,273 @@
+from dagster._core.definitions.multi_dimensional_partitions import (
+    MultiPartitionKey,
+    MultiPartitionsDefinition,
+    get_multipartition_key_from_tags,
+)
+from dagster._core.definitions.partition import PartitionsDefinition, PartitionsSubset
+from dagster._core.storage.tags import (
+    MULTIDIMENSIONAL_PARTITION_PREFIX,
+    get_dimension_from_partition_tag,
+)
+from dagster._serdes import deserialize_json_to_dagster_namedtuple, whitelist_for_serdes
+
+
+@whitelist_for_serdes
+class AssetStatusCacheValue(
+    NamedTuple(
+        "_AssetPartitionsStatusCacheValue",
+        [
+            ("latest_storage_id", int),
+            ("partitions_def_id", Optional[str]),
+            ("materialized_partition_subsets", Optional[str]),
+            # Future partition subset features will go here, e.g. stale partition subsets
+        ],
+    )
+):
+    """
+    Set of asset fields that reflect partition materialization status. This is used to display
+    global partition status in the asset view.
+
+    Properties:
+        latest_storage_id (int): The latest evaluated storage id for the asset.
+        partitions_def_id (Optional(str)): The serializable unique identifier for the partitions
+            definition. When this value differs from the new partitions definition, this cache
+            value needs to be recalculated. None if the asset is unpartitioned.
+        materialized_partition_subsets (Optional(str)): The serializable representation of the
+            materialized partition subsets, up to the latest storage id. None if the asset is
+            unpartitioned.
+    """
+
+    def __new__(
+        cls,
+        latest_storage_id: int,
+        partitions_def_id: Optional[str] = None,
+        materialized_partition_subsets: Optional[str] = None,
+    ):
+        check.int_param(latest_storage_id, "latest_storage_id")
+        check.opt_inst_param(
+            materialized_partition_subsets,
+            "materialized_partition_subsets",
+            str,
+        )
+        check.opt_str_param(partitions_def_id, "partitions_def_id")
+        return super(AssetStatusCacheValue, cls).__new__(
+            cls, latest_storage_id, partitions_def_id, materialized_partition_subsets
+        )
+
+    @staticmethod
+    def from_db_string(db_string):
+        if not db_string:
+            return None
+
+        cached_data = deserialize_json_to_dagster_namedtuple(db_string)
+        if not isinstance(cached_data, AssetStatusCacheValue):
+            return None
+
+        return cached_data
+
+    def deserialize_materialized_partition_subsets(
+        self, partitions_def: PartitionsDefinition
+    ) -> PartitionsSubset:
+        if not self.materialized_partition_subsets:
+            return partitions_def.empty_subset()
+
+        return partitions_def.deserialize_subset(self.materialized_partition_subsets)
+
+
+def get_materialized_multipartitions_from_tags(
```

Comment:
Nitpick: I think could leave `from_tags` out, because that's an implementation detail. This name led me to think that the tags would be one of the arguments.